consumer/src/main/java/com/flipkart/varadhi/consumer/UnGroupedMessageSrc.java

```diff
@@ -0,0 +1,50 @@
+package com.flipkart.varadhi.consumer;
+
+import com.flipkart.varadhi.entities.Offset;
+import com.flipkart.varadhi.spi.services.Consumer;
+import com.flipkart.varadhi.spi.services.PolledMessage;
+import com.flipkart.varadhi.spi.services.PolledMessages;
+import lombok.RequiredArgsConstructor;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.LinkedBlockingQueue;
+
+@RequiredArgsConstructor
+public class UnGroupedMessageSrc<O extends Offset> implements MessageSrc {
+
+    final Consumer<O> consumer;
+
+    private final BlockingQueue<MessageTracker> buffer = new LinkedBlockingQueue<>();
+
+    @Override
+    public CompletableFuture<Integer> nextMessages(MessageTracker[] messages) {
+        int offset = fetchFromBuffer(messages);
+        if (offset < messages.length) {
```

Comment:
Currently the impl is trying to fill the array completely. Though that is not strictly required, we should be able to return partial filled array.

Reply:
we can augment the consumer receive async future with `.orTimeout` so that in case no messages are available from the consumer we can still return early rather than waiting for a long time for messages.

Reply:
isn't that redundant, when you can return even early knowing that you had to call consumer for its (time taking) async operation?

Reply:
I think yes we can revert back to the previous behaviour where we return early once we fetch from the buffer